cql3/statements/select_statement.cc

```diff
@@ -772,8 +785,10 @@ select_statement::process_results(foreign_ptr<lw_shared_ptr<query::result>> resu
     const bool restrictions_need_filtering = _restrictions->need_filtering();
     const bool fast_path = !needs_post_query_ordering() && _selection->is_trivial() && !restrictions_need_filtering;
     if (fast_path) {
+        results->ensure_counts(); // Should be no-op since 2.1
```

Comment:
question about the comment: did you add this call here just to be extra sure that everything works, or did something break in your tests without it?

Reply:
Just to be sure. There is a call to ensure_counts() just below in the same method. If we want to get rid of this, I think it better be a separate commit to reduce risk.